maintenance_request_time_performance/models/maintenance.py

```diff
@@ -0,0 +1,151 @@
+            req.start_request_date = \
+                (req.start_request_datetime.date()
+                 if req.start_request_datetime else None)
+
+    def _compute_start_request_datetime_required(self):
+        for req in self:
+            req.start_request_datetime_required = \
+                req.stage_id.done or req.stage_id.in_progress
+
+    @api.depends('end_request_datetime')
+    def _compute_end_request_date(self):
+        for req in self:
+            req.end_request_date = \
+                (req.end_request_datetime.date()
+                 if req.end_request_datetime else None)
+
+    def _compute_end_request_datetime_required(self):
+        for req in self:
+            req.end_request_datetime_required = req.stage_id.done
+
+    @api.depends('start_request_datetime', 'end_request_datetime')
+    def _compute_real_request_duration(self):
+        for req in self:
+            total_hours = 0
+            if req.start_request_datetime and req.end_request_datetime \
+                    and req.start_request_datetime < req.end_request_datetime:
+                diff = req.end_request_datetime - req.start_request_datetime
+                total_hours = diff.total_seconds() / 3600
+            req.real_request_duration = total_hours
+
+    @api.depends('duration', 'real_request_duration')
+    def _compute_duration_difference(self):
+        for req in self:
+            difference = 0
+            if req.duration and req.real_request_duration:
+                difference = req.real_request_duration - req.duration
+            req.duration_difference = difference
+
+    @api.depends('schedule_date',
+                 'duration',
+                 'start_request_datetime',
+                 'end_request_datetime')
+    def _compute_completion_delay(self):
+        for req in self:
+            diff = None
+            if req.end_request_datetime:
+                schedule_dt = (req.schedule_date
+                               or req.start_request_datetime
+                               or req.create_date)
+                if schedule_dt.hour == 0 \
+                        and schedule_dt.minute == 0 \
+                        and schedule_dt.second == 0 \
+                        and req.start_request_datetime:
+                    # if has no time we put the start_request time
+                    schedule_dt = schedule_dt.replace(
+                        hour=req.start_request_datetime.hour,
+                        minute=req.start_request_datetime.minute,
+                        second=req.start_request_datetime.second)
+                if req.duration:
+                    # add theoretical duration to schedule_date
+                    schedule_dt = schedule_dt + timedelta(hours=req.duration)
+                diff = req.end_request_datetime - schedule_dt
+            delay = None
+            if diff:
+                delay = diff.total_seconds() / 3600
+            req.completion_delay = delay
```

Comment:
Alternative code (only suggestion, not checked):

```suggestion
            req.completion_delay = diff and (diff.total_seconds() / 3600)
```

Reply:
ok. I like it. I'll check it